### PR TITLE
[csharp-netcore] adds support for specifying cookies in the configuration object

### LINKED ISSUE #6191: [csharp-netcore] No provision to pass cookies to the ApiClient
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] What's the version of OpenAPI Generator used?
- [ ] Have you search for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

Their is a provision to pass headers in the configuration using DefaultHeaders. But passing `cookie` here does not seem to work.

For the below code, a wire capture shows the `origin` header, but `cookie` param is not found in the message. It gets saved into DefaultParameters but not consumed.

```
Org.OpenAPITools.Client.Configuration config = new Org.OpenAPITools.Client.Configuration();
var api = new Org.OpenAPITools.Api.NtpApi(config);
var url = endpoint + "ntp/Policies";
var intersightHeaders = new Dictionary<string, string>()
{
    { "cookie", "Param="+Value },
    { "origin", url }
};
foreach (var h in intersightHeaders)
{
    api.Configuration.DefaultHeaders.Add(h.Key, h.Value);
}
```

##### openapi-generator version
latest master

##### Suggest a fix

We could make a change in the ApiClient, if it comes across a header with Name "Cookie", then it could do the below instead.

```
if (configuration.DefaultHeaders != null)
{
    foreach (var headerParam in configuration.DefaultHeaders)
    {
        if (String.Equals(headerParam.Key, "cookie", StringComparison.OrdinalIgnoreCase))
        {
            string[] cookieSeparators = new string[] {";"};
            string[] EqualSeparator = new string[] {"="};
            string[] rawCookieParams = headerParam.Value.Split(cookieSeparators, StringSplitOptions.None);
            foreach (string param in rawCookieParams)
            {
                String[] ParamNameAndValue = param.Split(EqualSeparator, StringSplitOptions.None);
                if (ParamNameAndValue.Length == 2)
                {
                    request.AddCookie(ParamNameAndValue[0], ParamNameAndValue[1]);
                }
            }
            continue;
        }
        request.AddHeader(headerParam.Key, headerParam.Value);
    }
}
```

### PR DESCRIPTION
**Pull request by vvb**
fixes #6191 

Allows to specify cookies in DefaultHeaders. It takes care of specifying multiple `;` separated `Key=Value` cookies.
```
Org.OpenAPITools.Client.Configuration config = new Org.OpenAPITools.Client.Configuration();
var api = new Org.OpenAPITools.Api.NtpApi(config);
var url = endpoint + "ntp/Policies";
var headers = new Dictionary<string, string>()
{
    { "cookie", "k1=v1;k2=v2;v3"},
    { "origin", url }
};
foreach (var h in headers)
{
    api.Configuration.DefaultHeaders.Add(h.Key, h.Value);
}
```

/cc @wing328 @jimschubert 

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache [hunk 1]
```diff
@@ -271,6 +271,21 @@ namespace {{packageName}}.Client
             {
                 foreach (var headerParam in configuration.DefaultHeaders)
                 {
+                   if (String.Equals(headerParam.Key, "cookie", StringComparison.OrdinalIgnoreCase))
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache:274]: @vvb thanks for the PR but looks like this line is not indented correctly

Can you also update the petstore samples as well?

**Comment by wing328** [comment]
I think I may have a better approach. Let's sync up later today or tomorrow to discuss cookie parameter support in C# client.

File: modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache (hunk 1 above)
Comment (vvb) [reply modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache]: Thank you for catching this. I have updated the code.

**Comment by wing328** [comment]
I've filed https://github.com/OpenAPITools/openapi-generator/pull/6254 to add Cookie authentication support to the C# netcore client.

**Comment by wing328** [comment]
Closed via https://github.com/OpenAPITools/openapi-generator/pull/6254

Let me know if you encounter any issue with the cookie authentication in the C# netcore client

**Pull request closed without merging**